src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -152,7 +152,9 @@ private void AddSource(string fileName, string source, bool isRootContextDef = f
                 bool isInGlobalNamespace = @namespace == JsonConstants.GlobalNamespaceValue;
 
                 StringBuilder sb = new(@"// <auto-generated/>
-#nullable enable
+
+#nullable enable annotations
+#nullable disable warnings
```

Comment:
Are we ever generating public API? With annotations enabled we will still be outputting nullability metadata, and if we have reason to believe that the annotations are incorrect, that'll mean incorrect nullability information surfaced to consumers of the APIs.

If this is all internal to the assembly, it's probably fine.  If it's not, we might instead consider doing:
```C#
#nullable disable
#pragma warning disable CS8632
```
instead, or something along those lines.

Reply:
@eiriktsarpalis do you want to resolve this question before I hit merge?

Reply:
Yes let's please resolve this q first.

Reply:
we do produce some public APIs - AFAIK they're correct though

Reply:
> they're correct though

Ok